### DISCUSSION TITLE
Best practices to handle a Modal that needs fetched data in nextjs 14

### DISCUSSION THREAD

**Discussion by J4v4Scr1pt** [vercel/next.js#62734] [Help]
### Summary

If you have modal that should be shown in an nextjs 14 application, how would you approach that? 

We want to fetch as much data on the server as we can, but the content in the modal data should only be shown when the user press the modal button.

We have a couple of options:

[Intercepting routes](https://nextjs.org/docs/app/building-your-application/routing/intercepting-routes#modals): but this will not work well as just a standalone simple modal?
[UrlParamsState](https://www.youtube.com/watch?v=fwq9vePfwkI): Feels a bit hacky imo..
Fetch data on client using tanstack/serveraction: no fetch on server :(

I would like to understand if there is any best practices around building a modal in nextjs that needs data initially.
How do you handle a modal that needs fetched data?

**Comment by enzzzooo** [comment]
any idea?

**Reply by J4v4Scr1pt** [reply]
atm we use Tanstack and fetch it as soon as it opens. Server is good but sometimes it's just ok to do a classic fetch on client :). That's where we at now at least. Have not revised this since I posted the question :).

**Reply by enzzzooo** [reply]
Ok thanks! I am doing the same thing but running into old data when I change DB from a server action and call revalidatePath(). But since my fetch is on client the revalidatePath() doesn't work, at least in my code, any ideas?

**Reply by J4v4Scr1pt** [reply]
Maybe do `router.refresh()` in the client component?
https://nextjs.org/docs/app/api-reference/functions/use-router

> router.refresh(): Refresh the current route. Making a new request to the server, re-fetching data requests, and re-rendering Server Components. The client will merge the updated React Server Component payload without losing unaffected client-side React (e.g. useState) or browser state (e.g. scroll position).
